receiver/stanzareceiver/config.go

```diff
@@ -15,26 +15,41 @@
 package stanzareceiver
 
 import (
-	"github.com/open-telemetry/opentelemetry-log-collection/pipeline"
+	"github.com/open-telemetry/opentelemetry-log-collection/operator"
+	"go.opentelemetry.io/collector/config/configmodels"
 	"gopkg.in/yaml.v2"
 )
 
-// OperatorConfig is an alias that allows for unmarshaling outside of mapstructure
+// BaseConfig is the common configuration of a stanza-based receiver
+type BaseConfig struct {
+	configmodels.ReceiverSettings `mapstructure:",squash"`
+	Parsers                       ParserConfigs `mapstructure:"parsers"`
```

Comment:
I am probably missing something. Is it not possible to declare `Parsers []operator.Config` here? Do we need the `decodeParserConfigs` function?

Reply:
The alias exists to provide an opportunity for us to use [operator.Config.UnmarshalYAML](https://github.com/open-telemetry/opentelemetry-log-collection/blob/4622fc64438e4f05767b60c0b5ac6441b8408302/operator/config.go#L70), which is called when `decodeParserConfigs` calls `yaml.Unmarshal`, because the type is called. 

This method contains the logic that unmarshals different types of operators to the associated configurations. In theory, I think what you are suggesting might work if we had an equivalent mapstructure decode hook.